scss/_pagination.scss

```diff
@@ -1,31 +1,44 @@
 .pagination {
+  // scss-docs-start pagination-css-vars
+  --#{$variable-prefix}pagination-padding: #{$pagination-padding-y $pagination-padding-x};
+  --#{$variable-prefix}pagination-color: #{$pagination-color};
+  --#{$variable-prefix}pagination-bg: #{$pagination-bg};
+  --#{$variable-prefix}pagination-border-width: #{$pagination-border-width};
+  --#{$variable-prefix}pagination-border-color: #{$pagination-border-color};
+  --#{$variable-prefix}pagination-border-radius: #{$pagination-border-radius};
+  // scss-docs-end pagination-css-vars
+
   display: flex;
   @include list-unstyled();
 }
 
 .page-link {
   position: relative;
   display: block;
-  color: $pagination-color;
+  padding: var(--#{$variable-prefix}pagination-padding);
+  font-size: var(--#{$variable-prefix}pagination-font-size, inherit);
+  color: var(--#{$variable-prefix}pagination-color);
   text-decoration: if($link-decoration == none, null, none);
-  background-color: $pagination-bg;
-  border: $pagination-border-width solid $pagination-border-color;
+  background-color: var(--#{$variable-prefix}pagination-bg);
+  border: var(--#{$variable-prefix}pagination-border-width) solid var(--#{$variable-prefix}pagination-border-color);
   @include transition($pagination-transition);
 
   &:hover {
+    --#{$variable-prefix}pagination-color: #{$pagination-hover-color};
+    --#{$variable-prefix}pagination-bg: #{$pagination-hover-bg};
+    --#{$variable-prefix}pagination-border-color: #{$pagination-hover-border-color};
```

Comment:
Regarding states overrides, the current implementation is quite verbose and results in more CSS than without custom properties.

I can't think about a better way for now.

On the Sass side, there's probably room for a mixin, to reduce redundancy a bit?